engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -235,6 +237,53 @@ Stateful doConvertText(
     }
   }
 
+  @Specialization(
+      guards = {
+        "!methods.hasFunctionalDispatch(_this)",
+        "!methods.hasSpecialDispatch(_this)",
+        "getPolyglotCallType(_this, symbol.getName(), interop) == CONVERT_TO_DATE"
+      })
+  Stateful doConvertDate(
+      VirtualFrame frame,
+      Object state,
+      UnresolvedSymbol symbol,
+      Object _this,
+      Object[] arguments,
+      @CachedLibrary(limit = "10") MethodDispatchLibrary methods,
+      @CachedLibrary(limit = "1") MethodDispatchLibrary dateDispatch,
+      @CachedLibrary(limit = "10") InteropLibrary interop,
+      @Cached("wrapDateInAtom(interop)") java.util.function.Function<Object,Atom> wrapDate
+  ) {
+    try {
+      Atom date = wrapDate.apply(_this);
+      Function function = dateDispatch.getFunctionalDispatch(date, symbol);
+      arguments[0] = date;
+      return invokeFunctionNode.execute(function, frame, state, arguments);
+    } catch (MethodDispatchLibrary.NoSuchMethodException e) {
+      throw new PanicException(
+          Context.get(this).getBuiltins().error().makeNoSuchMethodError(_this, symbol), this);
+    }
+  }
+
+  final java.util.function.Function<Object,Atom> wrapDateInAtom(InteropLibrary iop) {
+    CompilerAsserts.neverPartOfCompilation();
+    final Context ctx = Context.get(this);
+    ctx.ensureModuleIsLoaded("Standard.Base.Data.Time.Date");
+    Optional<Module> dateModule = ctx.findModule("Standard.Base.Data.Time.Date");
```

Comment:
An example for core Enso language vs. standard library interop. I hope @hubertp will be able to handle even this case in some cleaner way as part of his work on #3363

Reply:
Looks like he will!